litmus-portal/cluster-agents/subscriber/subscriber.go

```diff
@@ -33,37 +33,30 @@ func init() {
 	k8s.KubeConfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
 	flag.Parse()
 
-	isConfirmed, newKey, err := k8s.IsClusterConfirmed()
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	if isConfirmed == true {
-		clusterData["KEY"] = newKey
-	} else if isConfirmed == false {
-
-		bodyText, err := gql.ClusterConfirm(clusterData)
+	if clusterData["IS_CLUSTER_CONFIRMED"] == "false" {
+		clusterConfirmByte, err := gql.ClusterConfirm(clusterData)
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		var responseInterface types.Payload
-		err = json.Unmarshal(bodyText, &responseInterface)
+		var clusterConfirmInterface types.Payload
+		err = json.Unmarshal(clusterConfirmByte, &clusterConfirmInterface)
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		if responseInterface.Data.ClusterConfirm.IsClusterConfirmed == true {
-			clusterData["KEY"] = responseInterface.Data.ClusterConfirm.NewClusterKey
+		if clusterConfirmInterface.Data.ClusterConfirm.IsClusterConfirmed == true {
+			clusterData["ACCESS_KEY"] = clusterConfirmInterface.Data.ClusterConfirm.NewClusterKey
 			_, err = k8s.ClusterRegister(clusterData)
 			if err != nil {
 				log.Fatal(err)
 			}
-			log.Println("cluster confirmed")
+			log.Println(clusterData["CLUSTER_ID"] + " has been confirmed")
```

Comment:
We might want to set the cluster confirm boolean to true in clusterData, currently we might not be using it but later on if we ever use that we will get confirmed as false always when the gql server is first starts as we are not updating it explicitly.

Reply:
But, In configmap, every data should be in the string. If we put the boolean value in cm, then it will automatically convert it to boolean

Reply:
I am not talking about the cm I am talking about the clusterData variable in the code, you are not updating clusterData["IS_CLUSTER_CONFIRMED"] to true after doing the cluster confirm. It can be string that's not a problem

Reply:
Ohk, Done.